### PR TITLE
General clean up, spelling, import order, visibility etc

### PR DESCRIPTION
**Pull request by agubler**
**Type:** bug

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

General clean up in the codebase, includes making `invalidate` protected.

### CODE REVIEW

File: src/customElements.ts [hunk 1]
```diff
@@ -65,12 +65,12 @@ export interface CustomElementInitializer {
...
  * Describes a custom element.
  *
- * @property    {string}                                tagName         	The tag name to register this widget under. Tag names must contain a "-"
- * @property    {WidgetConstructor}                     widgetConstructor   A widget Constructor that will return the widget to be wrapped in a custom element
- * @property    {CustomElementAttributeDescriptor[]?}   attributes     		A list of attributes to define on this element
- * @property    {CustomElementPropertyDescriptor[]?}    properties      	A list of properties to define on this element
- * @property    {CustomElementEventDescriptor[]?}       events          	A list of events to expose on this element
- * @property    {CustomElementInitializer?}             initialization  	A method to run to set custom properties on the wrapped widget
+ * @property tagName             The tag name to register this widget under. Tag names must contain a "-"
+ * @property widgetConstructorA  widget Constructor that will return the widget to be wrapped in a custom element
```
Comment (dylans) [inline src/customElements.ts]: widgetConstructor (remove the A)

File: src/customElements.ts [hunk 2]
```diff
@@ -65,12 +65,12 @@ export interface CustomElementInitializer {
  *
  * Describes a custom element.
  *
- * @property    {string}                                tagName         	The tag name to register this widget under. Tag names must contain a "-"
- * @property    {WidgetConstructor}                     widgetConstructor   A widget Constructor that will return the widget to be wrapped in a custom element
- * @property    {CustomElementAttributeDescriptor[]?}   attributes     		A list of attributes to define on this element
- * @property    {CustomElementPropertyDescriptor[]?}    properties      	A list of properties to define on this element
- * @property    {CustomElementEventDescriptor[]?}       events          	A list of events to expose on this element
- * @property    {CustomElementInitializer?}             initialization  	A method to run to set custom properties on the wrapped widget
+ * @property tagName             The tag name to register this widget under. Tag names must contain a "-"
```
Comment (dylans) [inline src/customElements.ts]: Probably a silly question, but should tagName and widgetConstructor be listed here when they are not listed in the interface?

Comment (agubler) [inline src/customElements.ts]: I'm not sure I understand, they are on the interface...

Comment (dylans) [inline src/customElements.ts]: I failed to expand that section of the interface in GH. Ignore. :)

**Pull request merged**